xbmc/addons/interfaces/GUI/Window.cpp

```diff
@@ -280,6 +315,578 @@ bool Interface_GUIWindow::do_modal(void* kodiBase, void* handle)
 
   return true;
 }
+//@}
+
+/*!
+ * Window control functions
+ */
+//@{
+bool Interface_GUIWindow::set_focus_id(void* kodiBase, void* handle, int control_id)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  CGUIAddonWindow* pAddonWindow = static_cast<CGUIAddonWindow*>(handle);
+  if (!addon || !pAddonWindow)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIWindow::%s - invalid handler data (kodiBase='%p', handle='%p') on addon '%s'",
+                          __FUNCTION__, addon, pAddonWindow, addon ? addon->ID().c_str() : "unknown");
+    return false;
+  }
+
+  CGUIWindow *pWindow = dynamic_cast<CGUIWindow*>(g_windowManager.GetWindow(pAddonWindow->m_windowId));
```

Comment:
The path is a remnant of the beginning

https://github.com/xbmc/xbmc/blob/4e9cb43928abbddd5d683cb5d9d66c72b3bbf88e/xbmc/addons/AddonCallbacksGUI.cpp#L388_L391

I think is to confirm Window registered with `g_windowManager.Add(window);`.

But from my side it can also be removed.